File: Backend/HBOS/Agents/Runtime/AgentRuntime.ts
export class AgentRuntime {

private agents:any[] = [];

register(agent:any){

    this.agents.push(agent);

}

run(name:string, task:any){

    const agent = this.agents.find(a => a.name === name);

    if(!agent){
        return {
            status:"agent_not_found"
        };
    }

    return agent.execute(task);

}

list(){

    return this.agents;

}

}

File: Backend/AI_Runtime/autogen/agent_runtime.py
class AgentRuntime:

    def __init__(self):
        self.agents={}


    def register(self,agent):

        self.agents[agent.name]=agent


    def execute(self,name,task):

        agent=self.agents.get(name)

        if agent:
            return agent.execute(task)

        return None


    def list_agents(self):

        return list(self.agents.keys())
